### ISSUE TITLE
[Feature request] --min-free-space 100GB ; I just wrecked 21 out of 24 btrfs drives

### ISSUE THREAD

**Issue by Motophan** [madMAx43v3r/chia-plot-sink#4]
btrfs on zoned devices has some weird metadata issues where the OS will report more data space than the drives can actually hold by a few GB due to empty space in metadata blocks being counted as free space.

If your plot just fits within 1GB it will actually not fit, overfill the drive, and nothing you do will be able to get the device mounted in a RW state to add a 2nd drive to it so you can balance metadata and delete 1 plot. 

Fixation: consider a drive full when freespace <= --min-free-space parameter in GB

**Comment by limsandy** [comment]
The way I see it is that's just a problem on your side when formatting your HDD. I formatted all my HDDs as ext4 and when there is 900 MB free space after copying the last plot file, I do want to fit the last plot file in there.
Maybe you can try setting some free unallocated free space with -m != 0 when formatting.

**Comment by limsandy** [comment]
-m >>> 0

**Comment by tjb-altf4** [comment]
+1 for an optional --min-free-space parameter

BTRFS does have a lot of advantages over regular EXT4 (including native support for HM-SMR disks), but it does unfortunately require balance operation(s) when it gets to that last ~150GB to squeeze in that last plot.

**Comment by Motophan** [comment]
@limsandy 
Hi, this is actually a specific HMSMR problem. You have 256MB zones and this causes the difference between reported space and actual space. You need to balance the drive to optimally fill these zones once it gets 99% full. 

This only occurs when the following two conditions are both true at the same time
1) user has a HM SMR drive
2) user is using btrfs in zoned mode ( drive does not have a CMR mode. )

Regular drives (the hdd controller manages the SMR regions, or drive is CMR) do not have this issue. You have these drives so you dont have this problem. 

tldr I need to leave 100-200GB space on the drive. I want to set some switch like `--min-free-space 100GB` to consider the drive full when it has 100GB left and stop sending plots to it

All I want is a minimum free space switch so I can run a balance before this program wrecks ~ 300TB of plots.

**Comment by Amplificator** [comment]
Why not just use file system quotas?

**Comment by Motophan** [comment]
because  btrfs will lie to the host
